dot/sync/chain_sync_integration_test.go

```diff
@@ -452,8 +452,8 @@ func TestChainSync_validateResponse_Integration(t *testing.T) {
 func TestChainSync_validateResponse_firstBlock_Integration(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	cs := newTestChainSync(ctrl)
-	bs := new(mocks.BlockState)
-	bs.On("HasHeader", mock.AnythingOfType("common.Hash")).Return(false, nil)
+	bs := NewMockBlockState(ctrl)
+	bs.EXPECT().HasHeader(gomock.AssignableToTypeOf(common.Hash{})).Return(false, nil).AnyTimes()
```

Comment:
Try to find the number of times each get called.
Also is the argument predictable (aka always the same) here? It would be nice to have it directly here as the argument.